test/test_torch.py

```diff
@@ -14457,50 +14490,64 @@ def tmp(dtype, device):
     ('addbmm', '', _small_2d, lambda t, d: [_small_3d(t, d), _small_3d(t, d)],
         1e-1, 1e-1, 1e-4, _float_types_with_bfloat16),
     ('addbmm', 'scalar', _small_2d, lambda t, d: [_number(0.4, 2, t), _small_3d(t, d), _small_3d(t, d)],
-        1e-1, 1e-1, 1e-4, _float_types_with_bfloat16),
+        1e-1, 1e-1, 1e-4, _float_types_with_bfloat16, True,
+        [_wrap_assert_warns("addbmm_? is deprecated")]),
```

Comment:
So... I see a bunch of deprecations here, but I don't see any newly added tests for the non-deprecated versions (using them as kwargs, I assume?) Do those tests exist? I don't see them obviously here.

Reply:
Some of these are also covered by the other tests above.

The main issue is that test machinery in `generate_tensor_op_tests` only handles positional args. I could modify the `arg_ctor` to return a tuple `(args, kwargs)` and then we can test the non-deprecated signatures.

Reply:
Yeah. I think I've added a gadget like `t(0, 2, 3, val=3)` to support args and kwargs, instead of just specifying as tuples.